twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillPreparer.java

```diff
@@ -384,6 +408,24 @@ private void setEnv(String runnableName, Map<String, String> env, boolean overwr
     }
   }
 
+  private void saveLogLevels(LogEntry.Level level) {
+    level = level == null ? LogEntry.Level.INFO : level;
```

Comment:
Shouldn't default it to `INFO` at all. It should be treated the same as `null` is provided as the log level value as in the other `saveLogLevels` method, which will throw `IllegalArgumentException`. Also, rename the `level` parameter to `rootLogLevel` to be clear.